src/vehicle_part_location.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#ifndef CATA_SRC_VEHICLE_PART_LOCATION_H
+#define CATA_SRC_VEHICLE_PART_LOCATION_H
+
+#include <string>
+#include <string_view>
+
+#include "translation.h"
+#include "type_id.h"
+
+class JsonObject;
+
+class vpart_location
+{
+    public:
+        bool was_loaded = false;
+        vpart_location_id id;
+
+        translation name;
+        translation description;
+        // Highest Z is rendered, potentially excluding roof. -1 never renders.
+        int z_order = 0;
+        // Display order in vehicle interact display, lowest first
+        int list_order = 5;
```

Comment:
If you want these initialized to these values, you need to initialize them in `optional`. Initialization in the structure definition is ignored for `optional` (because it can be omitted).

Reply:
I don't understand why that would be necessary. If optional gets called and and the field is missing from the json, it will initialize the member to a default constructed instance of its type, which should honor these defaults I think?

Reply:
Can I use the same trick as with the variant, and just pass was_loaded=false into optional, so it won't re-construct a new int for that c++ field if it's missing from the json? I'm starting to feel like was_loaded should be called was_initialized...

Reply:
Yeah, it's a bit gross. It's a bet that avoiding UB is better than confusing the people writing code :)

Reply:
I would suggest `optional( jo, was_loaded, "name", foo, foo )` - initialize it with it's own value if not loaded.

Reply:
It's not `was_initialized`, though - it is true when the entity being loaded is a copy (via `copy-from`) of an existing entity. The data was already loaded, and is being loaded again as an option to overwrite or modify it.

Reply:
ok, taking the suggestion from above